File: lib/w/cms/post.ex
defmodule W.CMS.Post do
  use Ecto.Schema
  import Ecto.Changeset

  schema "posts" do
    field :body, :string
    field :published, :boolean, default: false
    field :published_at, :naive_datetime
    field :title, :string
    field :view_name, :string
    field :views, :integer
    many_to_many :tags, W.CMS.Tag, join_through: "post_tags"#, on_replace: :delete

    timestamps()
  end

  @doc false
  def changeset(post, attrs) do
    post
    |> cast(attrs, [:title, :body, :published, :view_name])
    |> validate_required([:title, :body, :published, :view_name])
    |> unique_constraint(:title)
    |> unique_constraint(:view_name)
    |> cast_assoc(:tags, required: true)
  end
end

File: wj/entity/src/main/java/me/gsmr/entity/post/Post.java
package me.gsmr.entity.post;

import lombok.Data;

import java.io.Serializable;
import java.sql.Date;

@Data
public class Post implements Serializable {
    private static final long serialVersionUID = 1L;

    private Long id;
    private String title;
    private String body;
//    private String renderedBody;
//    private Boolean published;
//    private Date publishedAt;
//    private String viewName;
//    private Long views;
}
